cockatrice/src/dlg_connect.cpp

```diff
@@ -22,12 +23,19 @@ DlgConnect::DlgConnect(QWidget *parent)
     previousHosts = new QComboBox(this);
     previousHosts->installEventFilter(new DeleteHighlightedItemWhenShiftDelPressedEventFilter);
 
+	connect(previousHosts, SIGNAL(currentIndexChanged(int)), this, SLOT(updatePorts(int)));
+	
+	QMap<QString, UserConnection_Information> tmp = settingsCache->servers().getAllInfo();
+	
+	
+	
+
     QStringList previousHostList = settingsCache->servers().getPreviousHostList();
```

Comment:
I think you don't need this list at all - the map should contain all the information necessary to populate the dialog